core/pipeline/label.go

```diff
@@ -36,10 +36,14 @@ func RootLabels() []Label {
 }
 
 // LoadRootLabels loads default Pipeline labels from a workdir.
-func LoadRootLabels(workdir string) {
+func LoadRootLabels(workdir string, engineName string) {
 	loadOnce.Do(func() {
 		defer close(loadDoneCh)
 		defaultLabels = loadRootLabels(workdir)
+		defaultLabels = append(defaultLabels, Label{
+			Name:  "dagger.io/engine",
+			Value: engineName,
+		})
```

Comment:
Could make sense to push this a level deeper into `loadRootLabels` so it's all in one place, but it's really splitting hairs.

Reply:
Good point, changed to that